src/cli.rs

```diff
@@ -216,41 +256,40 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         let response = admin_client.create_admin(request).await?;
         println!("{:?}", response.into_inner());
     } else {
-        let data_dir = matches.value_of("datadir").unwrap_or("./.sensei");
-        let node = matches.value_of("node").unwrap_or("admin");
-        let macaroon_path = format!("{}/{}/.ldk/admin.macaroon", data_dir, node);
-
-        println!("macaroon path: {:?}", macaroon_path);
+        let macaroon_hex_str = matches.value_of("macaroon").unwrap_or("");
 
-        let mut macaroon_file = File::open(macaroon_path)?;
-        let mut macaroon_raw = Vec::new();
-        let _bytes = macaroon_file.read_to_end(&mut macaroon_raw)?;
-        let macaroon_hex_str = hex_utils::hex_str(&macaroon_raw);
+        let token_str = matches.value_of("token").unwrap_or("");
 
-        let channel = Channel::from_static("http://0.0.0.0:3000")
-            .connect()
-            .await?;
         let macaroon = MetadataValue::from_str(&macaroon_hex_str)?;
         let admin_macaroon = macaroon.clone();
+        
+        let token = MetadataValue::from_str(&token_str)?;
 
-        let mut client = NodeClient::with_interceptor(channel, move |mut req: Request<()>| {
+        let mut client = NodeClient::with_interceptor(channel.clone(), move |mut req: Request<()>| {
             req.metadata_mut().insert("macaroon", macaroon.clone());
             Ok(req)
         });
 
-        let admin_channel = Channel::from_static("http://0.0.0.0:3000")
-            .connect()
-            .await?;
+        // This is only needed for a couple of queries.  Disabling pagination effectively by large get
+        let pagination = PaginationRequest {
```

Comment:
I generally prefer pull request documentation/explanation for potentially surprising things, rather than comments, since comments tend to be lies waiting to happen, but put the explanatory comment there just in case.  LMK if you'd like it out.